pandas/_libs/tslibs/period.pyx

```diff
@@ -1255,6 +1255,33 @@ cdef class _Period(object):
 
     @property
     def dayofyear(self):
+        """
+        Return the day of the year.
+
+        This attribute returns the day of the year on which the particular
+        date occurs.
+
+        Returns
+        -------
+        int
+        The day of year.
+
+        See Also
+        --------
+        Period.dayofweek : Return the day of week.
+        Period.daysinmonth : Return the days in that month.
+
```

Comment:
Add PeriodIndex.dayofyear

Reply:
Do you mean 'weekofyear' ? (dayofyear is the current docstring)